src/main/java/duke/main/DukeWorker.java

```diff
@@ -0,0 +1,42 @@
+package duke.main;
+
+import duke.command.Command;
+import duke.task.Task;
+import duke.utils.CommandParser;
+import duke.utils.Parser;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+
+import java.io.IOException;
+import java.util.List;
+
+public class DukeWorker {
+
+    private Storage storage;
+    private TaskList taskList;
+
+    public DukeWorker(String storagePath) {
+        storage = new Storage(storagePath);
+        taskList = new TaskList();
+    }
+
+    public void initializeWorker() {
+        try {
+            boolean fileExist = storage.checkFileExistence();
+            if (!fileExist) {
+                storage.createStorage();
+            } else {
+                String data = storage.loadStorage();
+                List<Task> tasks = Parser.storageToTask(data);
+                taskList = new TaskList(tasks);
+            }
+        } catch (IOException e) {
```

Comment:
Here, we can catch more exceptions if the lower-level classes do not - assuming you implemented your code via the Exceptions whenever an unexpected input was given/logic error was encountered.